index.js

```diff
@@ -1228,50 +1228,29 @@ RedisClient.prototype.eval = RedisClient.prototype.EVAL = function () {
     });
 };
 
+exports.createClient = function(arg0, arg1, options) {
+    if (typeof arg0 === 'object' || arg0 === undefined) {
+        options = arg0 || options;
+        return createClient_tcp(default_port, default_host, options);
+    }
+    if (typeof arg0 === 'number' || typeof arg0 === 'string' && arg0.match(/^\d+$/)){
+        return createClient_tcp(arg0, arg1, options);
+    }
+    if (typeof arg0 === 'string') {
+        options = arg1 || {};
 
-exports.createClient = function(arg0, arg1, arg2){
-    if( arguments.length === 0 ){
-
-        // createClient()
-        return createClient_tcp(default_port, default_host, {});
-
-    } else if( typeof arg0 === 'number' ||
-        typeof arg0 === 'string' && arg0.match(/^\d+$/) ){
-
-        // createClient( 3000, host, options)
-        // createClient('3000', host, options)
-        return createClient_tcp(arg0, arg1, arg2);
-
-    } else if( typeof arg0 === 'string' ){
-        var parsed = URL.parse(arg0, true, true),
-            options = (arg1 || {});
-
+        var parsed = URL.parse(arg0, true, true);
```

Comment:
I did not test every possibility and would like to add some before this is merged but I'm quite certain the behavior is just the same as it was before. There should not be any BC.

This code is the same as it was before in [line 1245] (https://github.com/NodeRedis/node_redis/pull/786/files?diff=unified#diff-168726dbe96b3ce427e7fedce31bb0bcL1245).